### PR TITLE
Add get_addresses_info to the internal wallet

### PR DESCRIPTION
**Pull request by rantan**
We needs to get a list of address information to check the addresses is suitable on some processes.
We're going to use this method on tracking library that needs to check the addresses are tracking address or not.

### CODE REVIEW

File: spec/glueby/internal/wallet/active_record_wallet_adapter_spec.rb [hunk 1]
```diff
@@ -322,6 +322,88 @@
...
+          script_pubkey: '76a914a23178289ed6e2a15bd760e7406f85b32d79c17888ac',
+          label: label,
+          purpose: purpose,
+          wallet_id: wallet.id
+        })
+      ]
+    end
+    let(:addresses) { ['15JL32ZJTEeNUT7Fs348errZ8xmavXXhLp', '1FnbjGaaHaitz9FwTpDq4Ss8AkeKpLB5YY'] }
+    let(:purpose) { 'receive' }
+    let(:label) { nil }
```
Comment (azuchi) [inline spec/glueby/internal/wallet/active_record_wallet_adapter_spec.rb:356]: `label` and `purpose` are preferable for different data.

**Review by azuchi — CHANGES_REQUESTED** [review]

**Comment by azuchi** [comment]
> We're going to use this method on tracking library that needs to check the addresses are tracking address or not.

In this context, a method that searches for keys by `label` or `purpose` may be more suitable in terms of throughput.

**Comment by rantan** [comment]
> In this context, a method that searches for keys by label or purpose may be more suitable in terms of throughput.

I think the current implementation is more suitable for the use case.
Getting addresses by label would return a lot of addresses that include unnecessary addresses to the validation. It would lead to a big latency.

File: spec/glueby/internal/wallet/active_record_wallet_adapter_spec.rb (hunk 1 above)
Comment (rantan) [reply spec/glueby/internal/wallet/active_record_wallet_adapter_spec.rb:356]: What kind of data do you mean?
Would you show me an example?

**Review by azuchi — APPROVED** [review]

**Pull request merged**